SourceryRuntime/Sources/Variable.swift

```diff
@@ -51,6 +51,20 @@ public final class Variable: NSObject, SourceryModel, Typed, Annotated {
         return attributes[Attribute.Identifier.final.name] != nil
     }
 
+    /// Reference to type name where the variable is defined,
+    /// nil if defined outside of any `enum`, `struct`, `class` etc
+    public let definedInTypeName: TypeName?
+
+    /// Actual type name where the variable is defined if declaration uses typealias, otherwise just a `definedInTypeName`
+    public var actualDefinedInTypeName: TypeName? {
+        return definedInTypeName?.actualTypeName ?? definedInTypeName
```

Comment:
I may need some clarification of how the whole logic works. Initially I resolved the `definedInType` in the `resolve...` methods. But it lead to a problem with the following scenario:
```
class Foo {
  func bar() {}
}
extension Foo {
  func baz() {}
}
```
So that code above will end up being one type in `uniqueTypes` array. As Sourcery iterates through the `uniqueTypes` to resolve variable and method types - the definedInType for `baz` was resolved into `Foo` type which is not extension. But it is needed to be extension. That was the main reason to iterate through not unique types and assign correct (from my POV) types to `definedInType` property.

@ilyapuchka does it make sense?